gulpfile.js

```diff
@@ -1,6 +1,44 @@
 var gulp = require('gulp'),
+    jshint = require('gulp-jshint'),
+    concat = require('gulp-concat'),
+    uglify = require('gulp-uglify'),
     readme = require('gulp-readme-to-markdown');
 
+var config = {
+  jsPath: './src/js',
+  jsMinPath: './static/js',
+};
+
+gulp.task('jshint', function() {
+  gulp.src(config.jsPath + '/*.js')
+    .pipe(jshint())
+    .pipe(jshint.reporter('fail'));
+});
+
+gulp.task('js-frontend', ['jshint'], function() {
+  var minified = [
```

Comment:
I set this up in anticipation of some front end javascript needing to be written for a degree-search style typeahead.